src/osd/OSD.cc

```diff
@@ -2869,11 +2865,8 @@ int OSD::init()
       goto out;
   }
 
-  class_handler = new ClassHandler(cct);
-  cls_initialize(class_handler);
```

Comment:
```
2019-07-27T13:28:11.859+0000 7fcbbb736700 10 _load_class lock from /usr/lib/rados-classes/libcls_lock.so
2019-07-27T13:28:11.859+0000 7fcbbb736700  0 _load_class could not open class /usr/lib/rados-classes/libcls_lock.so (dlopen failed): /usr/lib/rados-classes/libcls_lock.so: undefined
 symbol: _Z16cls_cxx_setxattrPvPKcPN4ceph6buffer7v14_2_04listE
2019-07-27T13:28:11.859+0000 7fcbbb736700 -1 osd.3 33 class lock open got (5) Input/output error
2019-07-27T13:28:11.859+0000 7fcbbb736700  1 -- [v2:172.21.6.124:6818/10456,v1:172.21.6.124:6819/10456] --> 172.21.6.124:0/3205049774 -- osd_op_reply(5 reshard.0000000000 [call lock.lock] v0'0 uv0 ondisk = -5 ((5) Input/output error)) v8 -- 0x55fac669f180 con 0x55fac5e9f200
```

see http://pulpito.ceph.com/kchai-2019-07-27_13:12:09-rados-wip-kefu-testing-2019-07-27-1103-distro-basic-mira/4159190/

my guess is that the whole `src/objclass/class_api.cc` is either optimized out by the linker. as we are not using any symbol exposed by it.

i think we can either add the `class_api.cc` symbols which are supposed used by plugins to a dynamic-list-file, so these symbols can be added to the symbol table of the ceph-osd executable.

or we need add the `__attribute__((used))` specifier to all those functions.

see https://stackoverflow.com/questions/36803328/c-plugin-system-dlopen-fails?rq=1